accelerator/core/mam_core.h

```diff
@@ -25,81 +25,109 @@ extern "C" {
+ * @param bundle[in,out] The bundle that will be renewed
+ *
+ */
+void bundle_transactions_renew(bundle_transactions_t** bundle);
+
+/**
+ * @brief Initialize a mam_api_t object
+ *
+ * @param api[in,out] The MAM API object
+ * @param iconf[in] IOTA API parameter configurations
+ * @param seed[in] Seed of MAM channels. It is an optional choice
  *
  * @return return code
  */
-status_t ta_mam_init(mam_api_t* const api, const iota_config_t* const iconf, tryte_t const* const seed,
-                     mam_psk_t_set_t* const psks, mam_ntru_pk_t_set_t* const ntru_pks, tryte_t const* const psk,
-                     tryte_t const* const ntru_pk);
+status_t ta_mam_init(mam_api_t* const api, const iota_config_t* const iconf, tryte_t const* const seed);
+
+/**
+ * @brief Append Pre-Shared Key or NTRU public key into Pre-Shared Key set or NTRU public key set, respectively.
+ *
+ * @param psks[in] Pre-Shared Key set
+ * @param ntru_pks[in] NTRU public key set
+ * @param psk[in] List of Pre-Shared Key
+ * @param ntru_pk[in] List of NTRU public key
+ *
+ * @return status code
+ */
+status_t ta_set_mam_key(mam_psk_t_set_t* const psks, mam_ntru_pk_t_set_t* const ntru_pks, UT_array const* const psk,
+                        UT_array const* const ntru_pk);
 
 /**
  * @brief Write payload to bundle on the smallest secret key.
  *
  * With given channel_depth and endpoint_depth, generate the corresponding Channel ID and Endpoint ID, and write the
  * payload to a bundle. The payload is signed with secret key which has the smallest ordinal number.
  *
- * @param service - IRI node end point service[in]
- * @param api - The API [in,out]
- * @param channel_depth - Depth of channel merkle tree[in]
- * @param endpoint_depth - Depth of endpoint merkle tree[in]
- * @param psks - Pre-Shared Key set[in]
- * @param ntru_pks - NTRU public key set[in]
- * @param payload - The message that is going to send with MAM[in]
- * @param chid - Channel ID[out]
- * @param epid - Endpoint ID[out]
- * @param bundle - The bundle contains the Header and Packets of the current Message[out]
- * @param msg_id - Unique Message identifier under each channel[out]
- * @param ch_remain_sk - The number of MSS channel secret keys that haven't been used[out]
- * @param ep_remain_sk - The number of MSS endpoint secret keys of that haven't been used[out]
+ * @param service[in] IRI node end point service
+ * @param api[in,out] The MAM API object
+ * @param channel_depth[in] Depth of channel merkle tree
+ * @param psks[in] Pre-Shared Key set
+ * @param ntru_pks[in] NTRU public key set
+ * @param payload[in] The message that is going to send with MAM
+ * @param chid[out] Channel ID
+ * @param bundle[out] The bundle contains the Header and Packets of the current Message
+ * @param msg_id[out] Unique Message identifier under each channel
  *
  * @return return code
  */
 status_t ta_mam_written_msg_to_bundle(const iota_client_service_t* const service, mam_api_t* const api,
```

Comment:
Shall we manipulate these parameters within one function call? Too many parameters and harder to review.

Reply:
I will tryte to make this part concise in the future PRs.